swan-lake/resources/featured-scenarios/write-a-restful-api-with-ballerina.md

```diff
@@ -127,103 +187,77 @@ service /covid/status on new http:Listener(9000) {
 In this code:
 
 - Unlike normal functions, resource methods can have accessors. In this case, the accessor is set to `get`, which means only HTTP `GET` requests could hit this resource. Ballerina automatically serializes Ballerina records as JSON and sends them over the wire. 
-- The default HTTP response status code for a resource method other than `post` is `200 OK`. For an HTTP `POST` resource, the default HTTP response status code is `201 Created`. 
+- The default HTTP response status code for a resource method other than `post` is `200 OK`. The default HTTP response status code for an HTTP `POST` resource is `201 Created`. 
 
 ### Create the second resource to add data
 
-To create the second resource of the first endpoint to add new COVID-19 data to the dataset by ISO code, add the code below to the API template file (i.e., `service.bal`).
+Create the second resource of the first endpoint to add new data to the dataset by ISO code, using the Ballerina HTTP API Designer in VS Code, as shown below.
+
+> **Note:** Define an HTTP resource that allows the `POST` operation on the resource path `countries` and accepts a `CovidEntry[]` type payload named `covidEntries`, create a `201 Created` response with the `CovidEntry[]` as the body type, and also create a `409 Conflict` response with `string` as the body type by generating a `IsoCodeConflict` record type of the `http:Conflict` status code response.
```

Comment:
```suggestion
> **Note:** Define an HTTP resource that allows the `POST` operation on the resource path `countries` and accepts a `CovidEntry[]`-typed parameter named `covidEntries`.
```

> create a `201 Created` response with the `CovidEntry[]` as the body type, and also create a `409 Conflict` response with `string` as the body type by generating a `IsoCodeConflict` record type of the `http:Conflict` status code response.

These I'd make separate sentences
- it's more selecting the response payload type rather than creating a 201 response in the first case
- second one I would mention status code response and link to the BBE

Then, need to also say use these as response/return types

Also, payload type may be more appropriate than body type. Please check with Tharmi.

Reply:
```suggestion
> **Note:** Define an HTTP resource that allows the `POST` operation on the resource path `countries` and accepts a `CovidEntry[]`-typed payload parameter named `covidEntries`. Upon successful creation of covid entries, the resource should return the `CovidEntry[]` as the response payload. This can be done by selecting `CovidEntry[]` as the default(`201 Created`) response type. Additionally, in case of conflict entries, the resource should return a `409 Conflict` response with `string` payload. This can be done by generating a [status code response](/learn/by-example/http-send-different-status-codes-with-payload/) by selecting the response status code as `409 - conflict, the payload type as `string` and give `IsoCodeConflict` as the status code record name.

Reply:
```suggestion
> **Note:** Define an HTTP resource, which allows the `POST` operation on the resource path `countries` and accepts a `CovidEntry[]`-typed parameter named `covidEntries`, corresponding to the payload. Upon successful creation of the entries, the response shoud be a `201` response with payload of structure `CovidEntry[]`. This can be incorporated in the return type by selecting `CovidEntry[]` as the default (`201 Created`) response type. Additionally, in case of conflicting entries, the resource should return a `409 Conflict` response with a `string` payload. This can be done by generating a `409` [status code response](/learn/by-example/http-send-different-status-codes-with-payload/) by selecting the response status code as `409 - Conflict`, the payload type as `string`, and providing a name (`IsoCodeConflict` ) for the status code record.
```